glue/flagging_script_glue/flagging.py

```diff
@@ -767,131 +742,97 @@ def z_normalize_groupby(s: pd.Series):
-            (df["sv_pricing"].str.contains("Low")) & (df["sv_which_price"] == "(raw)"),
+            (df["sv_pricing"].str.contains("High") & (df["sv_which_price"] == "(raw)")),
+            (df["sv_pricing"].str.contains("Low") & (df["sv_which_price"] == "(raw)")),
+            (df["sv_pricing"].str.contains("High"))
+            & (df["sv_which_price"] == "(sqft)"),
             (df["sv_pricing"].str.contains("Low")) & (df["sv_which_price"] == "(sqft)"),
         ]
 
-        labels = [
-            "Home flip sale (high)",
-            "Family sale (high)",
-            "Non-person sale (high)",
-            "Anomaly (high)",
-            "High price swing",
-            "High price (raw & sqft)",
-            "High price (raw)",
-            "High price (sqft)",
-            "Home flip sale (low)",
-            "Family sale (low)",
-            "Non-person sale (low)",
-            "Anomaly (low)",
-            "Low price swing",
-            "Low price (raw & sqft)",
-            "Low price (raw)",
-            "Low price (sqft)",
+        # Define labels for price-based reasons
+        price_labels = [
+            "sv_ind_price_high_price_raw_and_sqft",
+            "sv_ind_price_low_price_raw_and_sqft",
+            "sv_ind_price_high_price",
+            "sv_ind_price_low_price",
+            "sv_ind_price_high_price_sqft",
+            "sv_ind_price_low_price_sqft",
         ]
 
-    df["sv_outlier_type"] = np.select(conditions, labels, default="Not outlier")
-
-    return df
+    combined_conditions = price_conditions + char_conditions
+    combined_labels = price_labels + char_labels
 
+    outlier_type_dict = dict(zip(combined_labels, combined_conditions))
 
-def outlier_flag(df: pd.DataFrame) -> pd.DataFrame:
-    """
-    Creates a flag that shows whether the record is an
-    outlier (a special flag) according to our outlier taxonomy.
-    Inputs:
-        df (pd.DataFrame): dataframe to create outlier flag
-    Outputs:
-        df (pd.DataFrame): dataframe with 'is_outlier' column
-    """
+    # Create indicator columns for each flag type
+    for label, condition in outlier_type_dict.items():
+        df[label] = condition.astype(int)
 
-    df["sv_is_outlier"] = np.select(
-        [(df["sv_outlier_type"] == "Not outlier")], [0], default=1
-    )
+    if not condos:
+        # Separate out combined raw and sqft label to be discrete cols
+        df.loc[
+            df["sv_ind_price_high_price_raw_and_sqft"] == 1,
+            ["sv_ind_price_high_price_sqft", "sv_ind_price_high_price"],
+        ] = 1
+        df.loc[
+            df["sv_ind_price_low_price_raw_and_sqft"] == 1,
+            ["sv_ind_price_low_price_sqft", "sv_ind_price_low_price"],
+        ] = 1
+
+        df.drop(
+            columns=[
+                "sv_ind_price_high_price_raw_and_sqft",
+                "sv_ind_price_low_price_raw_and_sqft",
+            ],
+            inplace=True,
+        )
```

Comment:
I didn't find a simple way to disentangle this from the prior functions so I had to place this logic here to separate raw and sqft classification.

Reply:
**[Question, non-blocking]** Hmm, I'm confused why this needs a separate step. Can't we just skip e.g. the `sv_ind_price_high_price_raw_and_sqft` condition and use two separate conditions to make this extra step unnecessary?

1.  `df["sv_ind_price_high_price"] = df["sv_pricing"].str.contains("High")`
2.  `df["sv_ind_price_high_price_sqft"] = df["sv_pricing"].str.contains("High") & df["sv_pricing"].str.contains("sqft")`